File: frontend/src/lib/api/task.ts
import apiClient from "@/lib/apiClient";

export interface UserTaskItem {
  id: string;
  title: string;
  description: string;
  category: string;
  difficulty: number;
  status: string;
  score: number | null;
  comment: string | null;
  completed_at: string | null;
  scored_at: string | null;
  date: string;
}

export interface TaskStats {
  xp: number;
  level: number;
}

export interface BabyAgeResponse {
  age_stage: string;
  source: string;
}

export function getDailyTasks(): Promise<UserTaskItem[]> {
  return apiClient.get("/api/v1/tasks/daily").then((r) => r.data);
}

export function completeTask(id: string): Promise<UserTaskItem> {
  return apiClient.post(`/api/v1/tasks/${id}/complete`).then((r) => r.data);
}

export function getPartnerTasks(): Promise<UserTaskItem[]> {
  return apiClient.get("/api/v1/tasks/partner").then((r) => r.data);
}

export function scoreTask(
  id: string,
  data: { score: number; comment?: string },
): Promise<UserTaskItem> {
  return apiClient.post(`/api/v1/tasks/${id}/score`, data).then((r) => r.data);
}

export function rejectTask(
  id: string,
  data?: { comment?: string },
): Promise<UserTaskItem> {
  return apiClient
    .post(`/api/v1/tasks/${id}/reject`, data ?? {})
    .then((r) => r.data);
}

export function getTaskStats(): Promise<TaskStats> {
  return apiClient.get("/api/v1/tasks/stats").then((r) => r.data);
}

export function getBabyAge(): Promise<BabyAgeResponse> {
  return apiClient.get("/api/v1/tasks/baby-age").then((r) => r.data);
}

export function setBabyAge(ageStage: string): Promise<void> {
  return apiClient
    .put("/api/v1/tasks/baby-age", { age_stage: ageStage })
    .then(() => undefined);
}

File: backend/internal/dto/task.go
package dto

import "time"

type UserTaskItem struct {
	ID          string     `json:"id"`
	Title       string     `json:"title"`
	Description string     `json:"description"`
	Category    string     `json:"category"`
	Difficulty  int        `json:"difficulty"`
	Status      string     `json:"status"`
	Source      string     `json:"source"`
	Score       *int       `json:"score"`
	Comment     *string    `json:"comment"`
	CompletedAt *time.Time `json:"completed_at"`
	ScoredAt    *time.Time `json:"scored_at"`
	Date        time.Time  `json:"date"`
}

type TaskScore struct {
	Score   int    `json:"score" binding:"required,min=1,max=5"`
	Comment string `json:"comment" binding:"max=500"`
}

type TaskStats struct {
	XP    int `json:"xp"`
	Level int `json:"level"`
}

type SetBabyAgeRequest struct {
	AgeStage string `json:"age_stage" binding:"required,oneof=pregnancy 0-3m 3-6m 6-12m 1-2y 2-3y 3-4y 4-5y"`
}

type BabyAgeResponse struct {
	AgeStage string `json:"age_stage"`
	Source   string `json:"source"` // "manual", "partner", "memory", ""
}
